### PR TITLE
Added real frequency to log 

### PR DESCRIPTION
**Pull request by tonynajjar**
---

## Basic Info

| Info | Please fill out this column |
| ------ | ----------- |
| Ticket(s) this addresses   | N/A |
| Primary OS tested on | Ubuntu |
| Robotic platform tested on | Own gazebo simulation |

---

## Description of contribution in a few bullet points

* I added logging of the real achieved frequency when the target is not reached. Context: we were getting the "Control loop missed" warning but wanted to know at what rate we are running to know over time if this gets worse.

It's unfortunate that rclcpp::WallRate doesn't offer a way to get the achieved period/frequency when sleep returns fals

## Description of documentation updates required from your changes

---

#### For Maintainers: 
- [ ] Check that any new parameters added are updated in navigation.ros.org
- [ ] Check that any significant change is added to the migration guide
- [ ] Check that any new features **OR** changes to existing behaviors are reflected in the tuning guide
- [ ] Check that any new functions have Doxygen added
- [ ] Check that any new features have test coverage
- [ ] Check that any new plugins is added to the plugins page
- [ ] If BT Node, Additionally: add to BT's XML index of nodes for groot, BT package's readme table, and BT library lists

### CODE REVIEW

**Comment by tonynajjar** [comment]
image

I sometimes get frequencies higher than the target one. Any idea why? Maybe because I'm using a separate clock?

**Review by SteveMacenski — APPROVED** [review]

File: nav2_controller/src/controller_server.cpp [hunk 1]
```diff
@@ -400,9 +403,14 @@ void ControllerServer::computeControl()
       }
 
       if (!loop_rate.sleep()) {
-        RCLCPP_WARN(
-          get_logger(), "Control loop missed its desired rate of %.4fHz",
-          controller_frequency_);
+        real_frequency =
```
Comment (SteveMacenski) [inline nav2_controller/src/controller_server.cpp:406]: This does not appear to be following the styling of the project (I see indents with 4 spaces, things being aligned pythonically, etc. 

Please lint this with ament_uncrustify / ament_cpplint for styling

**Review by SteveMacenski — CHANGES_REQUESTED** [review]

**Comment by tonynajjar** [comment]
on it

**Comment by tonynajjar** [comment]
I applied all your suggestions but still same result...
image

I commited the version with the ROS clock. I'm still investigating

**Comment by SteveMacenski** [comment]
It doesn't seem like this is something we should merge if it doesn't strictly speaking... work :frowning:

**Comment by tonynajjar** [comment]
I actually agree, unless the wallrate is actually unstable and this only exposes that. I doubt it though, will investigate a bit more tomorrow

**Comment by tonynajjar** [comment]
I can't find what's wrong and can't spend more time on it. I'm fine to close this PR if you want

**Comment by SteveMacenski** [comment]
OK :frowning_face: I understand.

**Pull request closed without merging**